blaze/expr/strings.py

```diff
@@ -110,7 +110,15 @@ def _dshape(self):
         '''
         since pandas supports concat for string columns, do the same for blaze
         '''
-        shape, schema = self.lhs.dshape.shape, DataShape(String())
+        shape = self.lhs.dshape.shape
+        if isinstance(self.lhs.schema.measure, Option):
+            schema = self.lhs.schema
+        elif isinstance(self.rhs.schema.measure, Option):
+            schema = self.rhs.schema
+        else:
+            # convert fixed length string to variable length string
+            schema = DataShape(String())
+
         return DataShape(*(shape + (schema,)))
```

Comment:
Fixed schema of output as follows:
- if `lhs` or `rhs` contain NULL values, then result will contain NULL
- if neither contain NULL values, then just return String() dshape